src/test/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcherTest.java

```diff
@@ -79,4 +92,349 @@ public void testShouldUpdate() throws IOException {
+        }
+      };
+
+      GitHubProject secondProject = new GitHubProject("test", "project2");
+
+      Runnable loadRepoTask = () -> {
+        try {
+          assertTrue(latch.await(10, TimeUnit.SECONDS));
+          fetcher.localRepositoryFor(secondProject);
+          checkLocalRepository(secondProject, 1);
+        } catch (InterruptedException | IOException e) {
+          throw new IllegalStateException("Load repository exception");
+        }
+      };
+
+      executorService.execute(cleanTask);
+      executorService.execute(loadRepoTask);
+    } finally {
+      executorService.shutdown();
+    }
+  }
+
+  @Test
+  public void testMultiThreadLocalRepoInfoTest() throws IOException, InterruptedException {
+    ExecutorService executorService = Executors.newFixedThreadPool(10);
+
+    GitHubProject project = new GitHubProject("test", "project");
+    Path projectDir = projectDir(project.name());
+    try (
+        Repository repository = FileRepositoryBuilder.create(projectDir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, project, "testFile");
+
+      fetcher.addRepositoryInfoForTesting(project, projectDir);
+
+      int numberOfThreads = 25;
+      CountDownLatch latch = new CountDownLatch(numberOfThreads);
+      for (int i = 0; i < numberOfThreads; i++) {
+        executorService.submit(() -> {
+          try {
+            localRepositoryFor(project);
+          } catch (IOException e) {
+            throw new IllegalStateException("Load repository exception");
+          }
+          latch.countDown();
+        });
+      }
+      latch.await();
+
+      checkLocalRepository(project, 1);
+    } finally {
+      executorService.shutdown();
+    }
+  }
+
+  @Test
+  public void testMultiThreadCleanUpTest() throws IOException, InterruptedException {
+    ExecutorService executorService = Executors.newFixedThreadPool(10);
+
+    GitHubProject project = new GitHubProject("test", "project");
+    Path projectDir = projectDir(project.name());
+    try (
+        Repository repository = FileRepositoryBuilder.create(projectDir.resolve(".git").toFile())) {
+      repository.create();
+
+      createRepoTestFile(repository, project, "testFile");
+
+      fetcher.addRepositoryInfoForTesting(project, projectDir);
+
+      localRepositoryFor(project, 1);
+
+      int numberOfThreads = 25;
+      CountDownLatch latch = new CountDownLatch(numberOfThreads);
+      for (int i = 0; i < numberOfThreads; i++) {
+        executorService.submit(() -> {
+          try {
+            cleanup(project);
+          } catch (IOException e) {
+            throw new IllegalStateException("Clean up exception");
+          }
+          latch.countDown();
+        });
+      }
+      latch.await();
+
+      checkCleanUp(project, 0);
+    } finally {
+      executorService.shutdown();
+    }
+  }
+
+  @Test
+  public void testLocalRepositoriesInfoJson() throws IOException {
```

Comment:
Does this test check the content of the JSON file?

Reply:
Updated here https://github.com/SAP/fosstars-rating-core/pull/246#discussion_r448768601

Reply:
It was this method which had `cleanup()`. As I mentioned all the tests are basically reloading from the `.json` file before the check. Hence, this explicit and duplicate test for the same did not make sense.

Referring to https://github.com/SAP/fosstars-rating-core/pull/246#discussion_r448865394